napari/_vispy/vispy_base_layer.py

```diff
@@ -151,3 +151,7 @@ def _reset_base(self):
         self._on_opacity_change()
         self._on_blending_change()
         self._on_matrix_change()
+
+    def _on_camera_move(self, event=None):
+        """Camera was moved."""
+        pass
```

Comment:
I'm surprised you don't also need to hook up `_on_camera_zoom` too? i.e. `camera.events.zoom`

Reply:
I guess the zoom knowledge goes through `layer._update_draw`. https://github.com/napari/napari/blob/v0.4.0/napari/_qt/qt_viewer.py#L694-L702

I think this is ok now, but curious how you think about that. In the long run we might want to get rid of that method/ reduce the number of places where this data flows.

Given considerations like "the layer should have no knowledge of the currently viewed slice" it makes sense to not rely on that method more and use the camera where we can

Reply:
Yeah I'd have to think about this more. Zoom seems to mostly change the center I guess. But maybe if you do a perfectly centered zoom we'll need to add zoom!

Reply:
Ha! Yeah, we may need this eventually